### DISCUSSION TITLE
Export CSS to file

### DISCUSSION THREAD

**Discussion by joneslloyd** [ben-rogerson/twin.macro#376] [Q&A] [answered]
Hello,

Do you know if (using Twin, Tailwind and Goober) it's possible to export the CSS to its own file for production?

That way, the CSS file can be external and referenced with a `<link>`, to reduce payload size of the main JS bundle.

With HTTP/2 multiplexing, the CSS file could then be downloaded in parallel to the main JS file.

I know that this isn't appropriate/possible for all scenarios (ie, with dynamic client-side apps where the component composition isn't known ahead of time), but for what I'm working on (an embeddable widget using Preact) an exported CSS file would be appropriate.

**Comment by ben-rogerson** [comment, marked as answer]
All of the current presets have a client runtime so I'm afraid it's not possible - yet.
Once twin gets it's [linaria](https://github.com/ben-rogerson/twin.macro/pull/248) preset then we'll have an external css file output. I'm in the middle of some documentation updates and I plan on moving over to this shortly afterwards.

[More about linaria](https://github.com/callstack/linaria)

**Reply by joneslloyd** [reply]
Thanks for the quick response!

Got it – That sounds great!
